### ISSUE TITLE
paddle 单机多卡训练为何问题不断？？

### ISSUE THREAD

**Issue by Hizhaoyuan** [PaddlePaddle/models#1520]

**Comment by guru4elephant** [comment]
请描述具体问题

**Comment by qianledan** [comment]
我运行官网github中分布训练的例子run_ps_mode.sh出现：
-----------  Configuration Arguments -----------
async_mode: False
batch_size: 32
checkpoint: None
class_dim: 1000
data_dir: ../data/ILSVRC2012
enable_ce: False
fp16: False
image_shape: 3,224,224
lr: 0.1
lr_strategy: piecewise_decay
model: DistResNet
model_category: models
model_save_dir: output
multi_batch_repeat: 1
num_epochs: 120
num_threads: 8
pretrained_model: None
reduce_strategy: allreduce
scale_loss: 1.0
skip_unbalanced_data: False
split_var: True
start_test_pass: 0
total_images: 1281167
update_method: pserver
use_gpu: True
with_mem_opt: False
------------------------------------------------
----------- Configuration envs -----------
ENV PADDLE_TRAINERS_NUM:2
ENV PADDLE_TRAINER_ID:1
ENV PADDLE_CURRENT_ENDPOINT:127.0.0.1:7161
ENV PADDLE_PSERVER_ENDPOINTS:127.0.0.1:7160,127.0.0.1:7161
ENV PADDLE_TRAINING_ROLE:TRAINER
------------------------------------------------
start lr: 0.1, end lr: 0.2, decay boundaries: [600570, 1201140, 1601520]
start lr: 0.1, end lr: 0.2, decay boundaries: [600570, 1201140, 1601520]
start lr: 0.1, end lr: 0.2, decay boundaries: [600570, 1201140, 1601520]
start lr: 0.1, end lr: 0.2, decay boundaries: [600570, 1201140, 1601520]
get_pserver_program() is deprecated, call get_pserver_programs() to get pserver main and startup in a single call.
get_pserver_program() is deprecated, call get_pserver_programs() to get pserver main and startup in a single call.
I0314 19:08:38.380228 29520 grpc_server.cc:430] Server listening on 127.0.0.1:7161 selected port: 7161
I0314 19:08:38.468055 29569 grpc_server.cc:430] Server listening on 127.0.0.1:7160 selected port: 7160
W0314 19:08:41.237395 28935 device_context.cc:263] Please NOTE: device: 0, CUDA Capability: 37, Driver API Version: 9.0, Runtime API Version: 9.0
W0314 19:08:41.237438 28935 device_context.cc:271] device: 0, cuDNN Version: 7.0.
Traceback (most recent call last):
  File "vl_dist_train.py", line 356, in <module>
    main()
  File "vl_dist_train.py", line 353, in main
    train_parallel(args)
  File "vl_dist_train.py", line 239, in train_parallel
    startup_exe.run(startup_prog)
  File "/home/qld/.pyenv/versions/3.6.7/lib/python3.6/site-packages/paddle/fluid/executor.py", line 525, in run
    use_program_cache=use_program_cache)
  File "/home/qld/.pyenv/versions/3.6.7/lib/python3.6/site-packages/paddle/fluid/executor.py", line 591, in _run
    exe.run(program.desc, scope, 0, True, True)
paddle.fluid.core.EnforceNotMet: Invoke operator fill_constant error.
Python Callstacks:
  File "/home/qld/.pyenv/versions/3.6.7/lib/python3.6/site-packages/paddle/fluid/framework.py", line 1382, in _prepend_op
    attrs=kwargs.get("attrs", None))
  File "/home/qld/.pyenv/versions/3.6.7/lib/python3.6/site-packages/paddle/fluid/initializer.py", line 167, in __call__
    stop_gradient=True)
  File "/home/qld/.pyenv/versions/3.6.7/lib/python3.6/site-packages/paddle/fluid/framework.py", line 1198, in create_var
    kwargs['initializer'](var, self)
  File "/home/qld/.pyenv/versions/3.6.7/lib/python3.6/site-packages/paddle/fluid/layer_helper.py", line 402, in set_variable_initializer
    initializer=initializer)
  File "/home/qld/.pyenv/versions/3.6.7/lib/python3.6/site-packages/paddle/fluid/optimizer.py", line 176, in _add_accumulator
    var, initializer=Constant(value=float(fill_value)))
  File "/home/qld/.pyenv/versions/3.6.7/lib/python3.6/site-packages/paddle/fluid/optimizer.py", line 515, in _create_accumulators
    self._add_accumulator(self._velocity_acc_str, p)
  File "/home/qld/.pyenv/versions/3.6.7/lib/python3.6/site-packages/paddle/fluid/optimizer.py", line 223, in _create_optimization_pass
    [p[0] for p in parameters_and_grads])
  File "/home/qld/.pyenv/versions/3.6.7/lib/python3.6/site-packages/paddle/fluid/optimizer.py", line 350, in apply_gradients
    optimize_ops = self._create_optimization_pass(params_grads)
  File "/home/qld/.pyenv/versions/3.6.7/lib/python3.6/site-packages/paddle/fluid/optimizer.py", line 405, in minimize
    optimize_ops = self.apply_gradients(params_grads)
  File "vl_dist_train.py", line 182, in build_program
    optimizer.minimize(avg_cost)
  File "vl_dist_train.py", line 217, in train_parallel
    train_pyreader, train_cost, train_acc1, train_acc5 = build_program(True, train_prog, startup_prog, args)
  File "vl_dist_train.py", line 353, in main
    train_parallel(args)
  File "vl_dist_train.py", line 356, in <module>
    main()
C++ Callstacks:
Enforce failed. Expected allocating <= available, but received allocating:10484135684 > available:1132330752.
Insufficient GPU memory to allocation. at [/paddle/paddle/fluid/platform/gpu_info.cc:216]
PaddlePaddle Call Stacks:
0       0x7f96ff8b7e0dp void paddle::platform::EnforceNotMet::Init<std::string>(std::string, char const*, int) + 365
1       0x7f96ff8b8157p paddle::platform::EnforceNotMet::EnforceNotMet(std::string const&, char const*, int) + 87
2       0x7f970137b8d8p paddle::platform::GpuMaxChunkSize() + 744
3       0x7f9701372089p
4       0x7f97a457da99p
5       0x7f970137185dp paddle::memory::legacy::GetGPUBuddyAllocator(int) + 109
6       0x7f9701372547p void* paddle::memory::legacy::Alloc<paddle::platform::CUDAPlace>(paddle::platform::CUDAPlace const&, unsigned long) + 39
7       0x7f9701372c15p paddle::memory::allocation::LegacyAllocator::AllocateImpl(unsigned long, paddle::memory::allocation::Allocator::Attr) + 389
8       0x7f9701374f1bp paddle::memory::allocation::Allocator::Allocate(unsigned long, paddle::memory::allocation::Allocator::Attr) + 27
9       0x7f97013666c3p paddle::memory::allocation::AllocatorFacade::Alloc(boost::variant<paddle::platform::CUDAPlace, paddle::platform::CPUPlace, paddle::platform::CUDAPinnedPlace, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_> const&, unsigned long, paddle::memory::allocation::Allocator::Attr) + 435
10      0x7f97013667e1p paddle::memory::allocation::AllocatorFacade::AllocShared(boost::variant<paddle::platform::CUDAPlace, paddle::platform::CPUPlace, paddle::platform::CUDAPinnedPlace, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_> const&, unsigned long, paddle::memory::allocation::Allocator::Attr) + 33
11      0x7f9701028320p paddle::memory::AllocShared(boost::variant<paddle::platform::CUDAPlace, paddle::platform::CPUPlace, paddle::platform::CUDAPinnedPlace, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_,boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_> const&, unsigned long, paddle::memory::allocation::Allocator::Attr) +48
12      0x7f9701323356p paddle::framework::Tensor::mutable_data(boost::variant<paddle::platform::CUDAPlace, paddle::platform::CPUPlace, paddle::platform::CUDAPinnedPlace, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>, paddle::framework::proto::VarType_Type, paddle::memory::allocation::Allocator::Attr, unsigned long) + 150
13      0x7f96ffd46101p paddle::operators::FillConstantKernel<float>::Compute(paddle::framework::ExecutionContext const&) const + 497
14      0x7f96ffd49123p std::_Function_handler<void (paddle::framework::ExecutionContext const&), paddle::framework::OpKernelRegistrarFunctor<paddle::platform::CUDAPlace, false, 0ul, paddle::operators::FillConstantKernel<float>, paddle::operators::FillConstantKernel<double>, paddle::operators::FillConstantKernel<long>, paddle::operators::FillConstantKernel<int>, paddle::operators::FillConstantKernel<paddle::platform::float16> >::operator()(char const*, char const*, int) const::{lambda(paddle::framework::ExecutionContext const&)#1}>::_M_invoke(std::_Any_data const&, paddle::framework::ExecutionContext const&) + 35
15      0x7f97012c5d03p paddle::framework::OperatorWithKernel::RunImpl(paddle::framework::Scope const&, boost::variant<paddle::platform::CUDAPlace, paddle::platform::CPUPlace, paddle::platform::CUDAPinnedPlace, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_> const&) const + 659
16      0x7f97012c3575p paddle::framework::OperatorBase::Run(paddle::framework::Scope const&, boost::variant<paddle::platform::CUDAPlace, paddle::platform::CPUPlace, paddle::platform::CUDAPinnedPlace, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_> const&) + 341
17      0x7f96ff9d441ap paddle::framework::Executor::RunPreparedContext(paddle::framework::ExecutorPrepareContext*, paddle::framework::Scope*, bool, bool, bool) + 218
18      0x7f96ff9d6415p paddle::framework::Executor::Run(paddle::framework::ProgramDesc const&, paddle::framework::Scope*, int, bool, bool) + 261
19      0x7f96ff89bf7bp
20      0x7f96ff8e2c9ep
21            0x4b071cp _PyCFunction_FastCallKeywords + 428
22            0x541558p
23            0x543ae7p _PyEval_EvalFrameDefault + 4871
24            0x5413f5p
25            0x5416a7p
26            0x5437e6p _PyEval_EvalFrameDefault + 4102
27            0x5413f5p
28            0x5416a7p
29            0x543ae7p _PyEval_EvalFrameDefault + 4871
30            0x540a51p
31            0x541901p
32            0x543ae7p _PyEval_EvalFrameDefault + 4871
33            0x540a51p
34            0x541901p
35            0x543ae7p _PyEval_EvalFrameDefault + 4871
36            0x5413f5p
37            0x542283p PyEval_EvalCode + 35
38            0x4277ffp PyRun_FileExFlags + 367
39            0x427a2cp PyRun_SimpleFileExFlags + 236
40            0x43ca35p Py_Main + 3445
41            0x41de12p main + 354
42      0x7f97a3ab5830p __libc_start_main + 240
43            0x41dee9p _start + 41

W0314 19:08:41.894523 28938 device_context.cc:263] Please NOTE: device: 0, CUDA Capability: 37, Driver API Version: 9.0, Runtime API Version: 9.0
W0314 19:08:41.894578 28938 device_context.cc:271] device: 0, cuDNN Version: 7.0.
F0314 19:09:09.338099 28677 grpc_client.cc:408] GetRPC name:[conv2d_0.w_0], ep:[127.0.0.1:7160], status:[-1] meets grpc error, error_code:4 error_message:Deadline Exceeded error_details:
*** Check failure stack trace: ***
    @     0x7f9a926e43fd  google::LogMessage::Fail()
    @     0x7f9a926e7eac  google::LogMessage::SendToLog()
    @     0x7f9a926e3f23  google::LogMessage::Flush()
    @     0x7f9a926e93be  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f9a93bab2da  paddle::operators::distributed::GRPCClient::Proceed()
    @     0x7f9b34aeac80  (unknown)
    @     0x7f9b372cb6ba  start_thread
    @     0x7f9b368f141d  clone
    @              (nil)  (unknown)

**Comment by qingqing01** [comment]
>  Enforce failed. Expected allocating <= available, but received allocating:10484135684 > available:1132330752.
> Insufficient GPU memory to allocation. at [/paddle/paddle/fluid/platform/gpu_info.cc:216]
PaddlePaddle Call Stacks:

<= 1.5.2 的版本建议设置:

```
export FLAGS_fast_eager_deletion_mode=1
export FLAGS_eager_delete_tensor_gb=0.0
```
